airbyte-integrations/connectors/destination-iceberg-v2/src/main/kotlin/io/airbyte/integrations/destination/iceberg/v2/io/IcebergUtil.kt

```diff
@@ -0,0 +1,175 @@
+import org.apache.iceberg.CatalogUtil.ICEBERG_CATALOG_TYPE
+import org.apache.iceberg.CatalogUtil.ICEBERG_CATALOG_TYPE_NESSIE
+import org.apache.iceberg.FileFormat
+import org.apache.iceberg.Schema
+import org.apache.iceberg.SortOrder
+import org.apache.iceberg.Table
+import org.apache.iceberg.TableProperties.DEFAULT_FILE_FORMAT
+import org.apache.iceberg.aws.s3.S3FileIO
+import org.apache.iceberg.catalog.Catalog
+import org.apache.iceberg.catalog.SupportsNamespaces
+import org.apache.iceberg.catalog.TableIdentifier
+import org.apache.iceberg.data.Record
+
+private val logger = KotlinLogging.logger {}
+
+/** Collection of Iceberg related utilities. */
+object IcebergUtil {
+    internal class InvalidFormatException(message: String) : Exception(message)
+
+    private val generationIdRegex = Regex("^ab-generation-id-\\d+\$")
+    fun assertGenerationIdSuffixIsOfValidFormat(generationId: String) {
+        if (!generationIdRegex.matches(generationId)) {
+            throw InvalidFormatException(
+                "Invalid format: $generationId. Expected format is 'ab-generation-id-<number>'",
+            )
+        }
+    }
+
+    fun constructGenerationIdSuffix(stream: DestinationStream): String {
+        return constructGenerationIdSuffix(stream.generationId)
+    }
+
+    fun constructGenerationIdSuffix(generationId: Long): String {
+        if (generationId < 0) {
+            throw IllegalArgumentException(
+                "GenerationId must be non-negative. Provided: ${generationId}",
+            )
+        }
+        return "ab-generation-id-${generationId}"
+    }
+    /**
+     * Builds an Iceberg [Catalog].
+     *
+     * @param catalogName The name of the catalog.
+     * @param properties The map of catalog configuration properties.
+     * @return The configured Iceberg [Catalog].
+     */
+    fun createCatalog(catalogName: String, properties: Map<String, String>): Catalog {
+        return CatalogUtil.buildIcebergCatalog(catalogName, properties, Configuration())
+    }
+
+    /**
+     * Builds (if necessary) an Iceberg [Table]. This includes creating the table's namespace if it
+     * does not already exist. If the [Table] already exists, it is loaded from the [Catalog].
+     *
+     * @param tableIdentifier The [TableIdentifier] that contains the [Table]'s namespace and name.
+     * @param catalog The Iceberg [Catalog] that contains the [Table] or should contain it once
+     * created.
+     * @param schema The Iceberg [Schema] associated with the [Table].
+     * @param properties The [Table] configuration properties derived from the [Catalog].
+     * @return The Iceberg [Table], created if it does not yet exist.
+     */
+    fun createTable(
+        tableIdentifier: TableIdentifier,
+        catalog: Catalog,
+        schema: Schema,
+        properties: Map<String, String>
+    ): Table {
+        if (
+            catalog is SupportsNamespaces && !catalog.namespaceExists(tableIdentifier.namespace())
```

Comment:
weird thought: maybe this method should take a DestinationStream.Descriptor instead of TableIdentifier?

if the catalog _doesn't_ support namespaces, then we (possibly?) want to `catalog.buildTable(TableIdentifier.of(null, stream.namespace + "_" + stream.name))`

(but also, nessie + glue both support namespaces, so we don't need to decide right now)

Reply:
First part done and agree on revisiting this when we implement a catalog that doesn't support namespaces.